### DISCUSSION TITLE
Idea: More context data (e.g. Commit Message, Commit Range)

### DISCUSSION THREAD

**Discussion by mooyoul** [community/community#26103] [Actions]
<p>It would be nice if gh-actions adds more context data.</p>
<p>My Wishlists are:</p>
<ul>
<li>Commit Message: This can be useful for conditional builds</li>
<li>Commit Range: This can be useful for linting commit message (e.g. <a href="https://github.com/conventional-changelog/commitlint" rel="nofollow noopener">CommitLint</a>)</li>
</ul>
<p>Also there is a CI Environment Variable comparision matrix between CI providers:</p>
<p><a href="https://github.com/pvdlg/env-ci#supported-ci" rel="nofollow noopener">https://github.com/pvdlg/env-ci#supported-ci</a></p>

**Comment by chrispat** [comment]
<p>GitHub Actions provides a very rich set of context data includeing the full event payload for the webhook that triggered the run.  For more information about how to access the context see <a href="https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions" rel="nofollow noopener">https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions</a>. </p>

**Comment by electricsheep** [comment]
<p>Is there a way of accessing the commit message from the event payload? I can’t find any information in the linked documentation</p>

**Comment by electricsheep** [comment, marked as answer]
<p>Answering my own question, you can access a commit message like this from a push event, where the commit array is all the commits included in a push:</p>
<pre><code>${{github.event.commits[0].message}}
</code></pre>
<p><a href="https://developer.github.com/v3/activity/events/types/#pushevent" rel="nofollow noopener">Found in the documentation here</a></p>
